### PR TITLE
Filter the no repeats fragments and add test stage

### PR DESCRIPTION
**Pull request by yangtcai**
I divide the dataset into train and test, so we can see how it performs on unseen sequences :D

### CODE REVIEW

**Comment by yangtcai** [comment]
BTW, pandas is really great, when we filter the no repeats subsequence only needs 5 min on the laptop, so it makes sense!!!!

File: dataloader.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import pathlib
 
 from typing import Union
+from tqdm import tqdm
```
Comment (williamstark01) [inline dataloader.py:5]: tqdm is a third party library

File: dataloader.py [hunk 2]
```diff
@@ -72,14 +65,39 @@ def forward_strand(self, index):
...
-            result_type="broadcast",
-        )
+        return repeats_in_sequence
+
+    def filter_no_repeat(self):
+
+        repeat_list = []
+        for index in tqdm(range(self.len // (self.segment_length - self.overlap))):
+            genome_index = index * (self.segment_length - self.overlap)
+            # produce sequence with overlap 500 of length 2000
```
Comment (williamstark01) [inline dataloader.py:75]: obsolete comment

**Review by williamstark01 — CHANGES_REQUESTED** [review]
Very good job, as you say, 5 minutes is not too long for preparing the dataset (even though it can probably be optimized even further).

And I guess you are running an experiment with the new test set, looking forward to see how well it performs.

File: dataloader.py [hunk 3]
```diff
@@ -255,14 +271,14 @@ def sequence_to_label_encoding(self, sequence):
...
-    #     print(f"{index=}, {item=}")
-    #     annotation = item[1]
-    #     if annotation["classes"].nelement() > 0:
-    #         break
-    dataloader = build_dataloader()
-    for data in dataloader:
-        print(data)
+    print(dataset[0])
+    while True:
+        index = random.randint(1, 5000)
```
[the remark this answers is not in the document]
Comment (yangtcai) [reply dataloader.py:276]: this index is the index in the repeat_list we filterd in the `__init__`, the repeat_list will store the original index, so when we call `__getitem__(self, index)`, we use `repeat_list[index]` to get the original index, the original index is the position of subsequene.

[the remark this answers is not in the document]
Comment (williamstark01) [reply dataloader.py]: Great, similar indexing with the one before filtering.
